src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -6905,4 +6905,83 @@ public object VisitParenExpression(ParenExpressionAst parenExpressionAst)
             return parenExpressionAst.Pipeline.Accept(this);
         }
     }
+
+
+    /// <summary>
+    /// Completes with the property names of the InputObject.
+    /// </summary>
+    public class PropertyNameCompleter : IArgumentCompleter
```

Comment:
I think this should be internal access.

Reply:
I understand from the perspective of this CL, but in general, isn't it cool if we could get completion for PropertyNames for third party cmdlets? 
Others (like me :) ) can make use of that too.

Reply:
I suggest to convert to internal for fast merging the PR and open a tracking issue to wait PowerShell Committee approval.

Reply:
It is now internal.